src/charm.py

```
@@ -179,6 +221,43 @@ def _on_provide_certificate_action(self, event: ActionEvent) -> None:
             return
         event.set_results({"result": "Certificates successfully provided."})
 
+    def _on_trust_certificate_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Provide trust certificate if configured to new requirer.
+
+        Args:
+            event: Juju event.
+
+        Returns:
+            None
+        """
+        try:
+            bundle = self._get_trusted_certificate_bundle()
+            self.certificate_transfer.add_certificates(
+                bundle,
+                relation_id=event.relation.id,
+            )
+        except ValueError as e:
+            logger.warning("Trust certificate relation cannot be fulfilled: %s", e)
+
+    def _get_trusted_certificate_bundle(self) -> Set[str]:
+        """Provide trust certificate if configured to new requirer.
+
+        Args:
+            event: Juju event.
+
+        Returns:
+            list[Certificate]: List of certificates in the bundle
+
+        Raises:
+            ValueError: if the configuration is not provided or invalid
+        """
+        if bundle := self.model.config.get("trusted-certificate-bundle"):
+            return {
+                cert.public_bytes(serialization.Encoding.PEM).decode("utf-8").strip()
+                for cert in parse_pem_bundle(str(bundle))
+            }
+        raise ValueError("Trusted certificate bundle config is not set")
```

I don't think this should raise a `ValueError`:

1. It's not the right exception (from the docs: "Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.")
2. [opinion] I don't think this config value not existing is exceptional. It seems like this would be the common case when it just hasn't been set yet, which seems likely if you just made the relation.

Can we just return `None` here? I actually think we go overboard with returning `None` in a lot of cases (where I think we should actually raise an exception) but in this case I feel like `None` makes sense. However, we might have a `try..except` around the parsing and decoding bits where it might make sense to raise an exception.

---

That case is now raising `KeyError`. `None` is the devil.